types/dompurify/index.d.ts

```diff
@@ -3,32 +3,63 @@
 // Definitions by: Dave Taylor <http://davetayls.me>, Samira Bazuzi <https://github.com/bazuzi>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-interface IDOMPurify {
-  sanitize(s:string):string;
-  sanitize(s:string, config:IDOMPurifyConfig):any;
+export = DOMPurify;
+export as namespace DOMPurify;
 
-  addHook(hook:string, cb:(currentNode:Element, data:any, config:IDOMPurifyConfig) => Element):void;
+declare var DOMPurify: DOMPurify;
+
+interface DOMPurify {
+    sanitize(source: string | Node): string;
+    sanitize(source: string | Node, config: DOMPurifyConfig & { RETURN_DOM_FRAGMENT?: false; RETURN_DOM?: false; }): string;
+    sanitize(source: string | Node, config: DOMPurifyConfig & { RETURN_DOM_FRAGMENT: true; }): DocumentFragment;
+    sanitize(source: string | Node, config: DOMPurifyConfig & { RETURN_DOM: true; }): HTMLElement;
+    sanitize(source: string | Node, config: DOMPurifyConfig): string | HTMLElement | DocumentFragment;
+    addHook(hook: 'uponSanitizeElement', cb: (currentNode: Element, data: DOMPurifySanitizeElementHookEvent, config: DOMPurifyConfig) => void): void;
+    addHook(hook: 'uponSanitizeAttribute', cb: (currentNode: Element, data: DOMPurifySanitizeAttributeHookEvent, config: DOMPurifyConfig) => void): void;
+    addHook(hook: DOMPurifyHookName, cb: (currentNode: Element, data: DOMPurifyHookEvent, config: DOMPurifyConfig) => void): void;
 }
 
-interface IDOMPurifyConfig {
-  ADD_ATTR?:string[];
-  ADD_TAGS?:string[];
-  ALLOW_DATA_ATTR?:boolean;
-  ALLOWED_ATTR?:string[];
-  ALLOWED_TAGS?:string[];
-  FORBID_ATTR?:string[];
-  FORBID_TAGS?:string[];
-  KEEP_CONTENT?:boolean;
-  RETURN_DOM?:boolean;
-  RETURN_DOM_FRAGMENT?:boolean;
-  RETURN_DOM_IMPORT?:boolean;
-  SAFE_FOR_JQUERY?:boolean;
-  SANITIZE_DOM?:boolean;
-  WHOLE_DOCUMENT?:boolean;
+interface DOMPurifyConfig {
+    ADD_ATTR?: string[];
+    ADD_TAGS?: string[];
+    ALLOW_DATA_ATTR?: boolean;
+    ALLOWED_ATTR?: string[];
+    ALLOWED_TAGS?: string[];
+    FORBID_ATTR?: string[];
+    FORBID_TAGS?: string[];
+    KEEP_CONTENT?: boolean;
+    RETURN_DOM?: boolean;
+    RETURN_DOM_FRAGMENT?: boolean;
+    RETURN_DOM_IMPORT?: boolean;
+    SAFE_FOR_JQUERY?: boolean;
+    SANITIZE_DOM?: boolean;
+    WHOLE_DOCUMENT?: boolean;
 }
 
-declare var DOMPurify:IDOMPurify;
+type DOMPurifyHookName
+    = 'beforeSanitizeElements'
+    | 'uponSanitizeElement'
+    | 'afterSanitizeElements'
+    | 'beforeSanitizeAttributes'
+    | 'uponSanitizeAttribute'
+    | 'afterSanitizeAttributes'
+    | 'beforeSanitizeShadowDOM'
+    | 'uponSanitizeShadowNode'
+    | 'afterSanitizeShadowDOM';
+
+type DOMPurifyHookEvent
+    = DOMPurifySanitizeElementHookEvent
+    | DOMPurifySanitizeAttributeHookEvent
+    | null;
```

Comment:
I want to get `DOMPurifyHookEvent: DOMPurifySanitizeElementHookEvent | DOMPurifySanitizeAttributeHookEvent | null` but `null` is removed by compiler. So I actually got `DOMPurifyHookEvent: DOMPurifySanitizeElementHookEvent | DOMPurifySanitizeAttributeHookEvent`. This type doesn't have `null`. How can I add to this type possibility to be `null`?